### PR TITLE
use state_machines dynamic state predicate methods

### PR DESCRIPTION
**Pull request by priyank-gupta**
refactor state_machine dynamic predicate methods for state.
Use new hash syntax in order_inventory_spec.

### CODE REVIEW

**Comment by Mafi88** [comment]
@priyank-gupta Could you please address houndci issues?

**Comment by priyank-gupta** [comment]
@Mafi88 Done!!!

File: core/spec/models/spree/order_inventory_spec.rb [hunk 1]
```diff
@@ -189,10 +193,12 @@
       end
 
       it 'only attempts to destroy as many units as are eligible, and return amount destroyed' do
-        allow(shipment).to receive_messages(inventory_units_for_item: [
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'shipped'),
-          mock_model(Spree::InventoryUnit, :variant_id => variant.id, :state => 'on_hand')
-        ])
+        allow(shipment).to receive_messages(
+          inventory_units_for_item: [
+            mock_model(Spree::InventoryUnit, variant_id: variant.id, state: 'shipped', shipped?: false),
```
Comment (Mafi88) [inline core/spec/models/spree/order_inventory_spec.rb]: @priyank-gupta Why have you added ```shipped?: false``` to these specs?
If ```state: 'shipped'```, isn't natural for ```shipped?``` to return true?. What am I missing here?

Comment (priyank-gupta) [inline core/spec/models/spree/order_inventory_spec.rb]: @Mafi88 As this is a mocked object, we need to specify the methods which are received by this object, in this case `shipped`. `state: 'shipped'` can only be used when `#state` is called on this object to return `'shipped'` whereas `#shipped?` will return `false`.
Also, I think here it should be `shipped?: true`. Fixing this.

**Comment by priyank-gupta** [comment]
@Mafi88 Fixed!!!

**Pull request merged**